frontend/src/metabase/components/EntityItem.jsx

```diff
@@ -105,17 +109,29 @@ function EntityItemMenu({
           event: `${analyticsContext};Entity Item;Archive Item;${item.model}`,
         },
       ].filter(action => action),
-    [item, onPin, onMove, onCopy, onArchive, analyticsContext],
+    [
+      showPinnedAction,
+      isPinned,
+      onPin,
+      analyticsContext,
+      item.model,
+      onMove,
+      onCopy,
+      onArchive,
+    ],
   );
   if (actions.length === 0) {
     return null;
   }
   return (
-    <EntityMenu
-      className={cx(className, "hover-child")}
-      triggerIcon="ellipsis"
-      items={actions}
-    />
+    <EntityMenuContainer align="center">
+      {!isPinned && <PinButton icon="pin" onClick={onPin} />}
```

Comment:
This could use a `Pin this` tooltip on hover.

Reply:
Done